src/VernaContextProvider.tsx

```diff
@@ -71,18 +84,76 @@ function VernaContextProvider({
   defaultUiSchema,
   defaultWidget,
   defaultReadOnly,
+  defaultSelector,
   children,
 }: PropsWithChildren<VernaContextProviderProps>) {
-  const [schema, setSchema] = useState<JSONSchema7>(defaultSchema);
-  const [uiSchema, setUiSchema] = useState<UiSchema>(
-    genDefaultUiSchema(defaultSchema, defaultUiSchema),
+  // The full schemas are not used by the lib itself but may be used for further implementation
+  const [fullSchema, setFullSchema] = useState<JSONSchema7>(defaultSchema || {});
+  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
+    genDefaultUiSchema(defaultSchema || {}, defaultUiSchema || {}),
   );
-  const [readOnly, setReadOnly] = useState<boolean>(defaultReadOnly);
-  const [widgets, setWidgets] = useState<WidgetType>(defaultWidget);
+  const [schema, _setSchema] = useState<JSONSchema7>(
+    defaultSelector &&
+      defaultSchema &&
+      defaultSchema['properties'] &&
+      defaultSchema.properties[defaultSelector]
+      ? (defaultSchema['properties'][defaultSelector] as JSONSchema7)
+      : defaultSchema || {},
+  );
```

Comment:
It's weird. We could simplify it :
- As defaultSchema has a default value
- By using optional chaining operator `?`
- By wrapping this condition within a function in order to improve readability

```suggestion
  const [schema, _setSchema] = useState<JSONSchema7>(
    defaultSelector && defaultSchema?.properties?.[defaultSelector]
      ? (defaultSchema['properties'][defaultSelector] as JSONSchema7)
      : defaultSchema,
  );
```